src/rpc/common/Validators.h

```diff
@@ -340,4 +344,49 @@ class CustomValidator final
     verify(boost::json::value const& value, std::string_view key) const;
 };
 
+// clang-format off
+/**
+ * @brief Provide a common used validator for ledger hash
+ * LedgerHash must be a string and hex
+ */
+inline CustomValidator LedgerHashValidator = validation::CustomValidator{
+    [](boost::json::value const& value, std::string_view key) -> MaybeError {
+        if(!value.is_string()){
+            return Error{RPC::Status{
+                RPC::RippledError::rpcINVALID_PARAMS, "ledgerHashNotString"}};;
+        }
+        ripple::uint256 ledgerHash;
+        if (!ledgerHash.parseHex(value.as_string().c_str()))
+            return Error{RPC::Status{
+                RPC::RippledError::rpcINVALID_PARAMS, "ledgerHashMalformed"}};
+        return MaybeError{};
+    }
+};
+
+/**
+ * @brief Provide a common used validator for ledger index
+ * LedgerIndex must be a string or int
+ * If it is string, it must be "validated" or "int"
+ */
+inline CustomValidator LedgerIndexValidator = validation::CustomValidator{
+    [](boost::json::value const& value, std::string_view key) -> MaybeError {
+        auto err = Error{RPC::Status{
+                    RPC::RippledError::rpcINVALID_PARAMS, "ledgerIndexMalformed"}};
+        if(!value.is_string() && !(value.is_uint64() || value.is_int64())){
+            return err;
+        }
+        if(value.is_string() && value.as_string()!="validated"){
+            auto str = value.as_string();
+            uint32_t unused;
+            auto [_, ec] { std::from_chars(str.data(), str.data() + str.size(), unused) };
+            if (ec != std::errc())
+            {
```

Comment:
I'd remove these curly braces - they don't help the code readability in this case and are ok to omit imo.